src/app/work-item/work-item-detail/work-item-detail.component.html

```diff
@@ -1,72 +1,87 @@
-    </div>
-    <br /><br />
-    <div class="form-group">
-      <label class="col-sm-1 control-label" for="wi-detail-type">Type</label>
-      <div class="col-sm-5">
-        <select [(ngModel)]="workItem.type" id="wi-detail-type" class="form-control">
-          <option *ngFor="let type of workItemTypes" [value]="type">{{type.substring(7,type.length)}}</option>
-        </select>
-      </div>
-    </div>
-    <br /><br />
-    <div class="form-group">
-      <label class="col-sm-1 control-label" for="wi-detail-creator">Creator</label>
-      <div class="col-sm-5">
-        <input type="text" [(ngModel)]="workItem.fields['system.creator']" id="wi-detail-creator" placeholder="Creator" class="form-control">
-      </div>
-    </div>
-    <div class="form-group">
-      <label class="col-sm-1 control-label" for="wi-detail-assignee">Assignee</label>
-      <div class="col-sm-5">
-        <input type="text" [(ngModel)]="workItem.fields['system.assignee']" id="wi-detail-assignee" placeholder="Assignee" class="form-control">
-      </div>
-    </div>
-    <div class="form-group">
-      <label class="col-sm-1 control-label" for="wi-detail-state">State</label>
-      <div class="col-sm-5">
-        <select [(ngModel)]="workItem.fields['system.state']" id="wi-detail-state" class="form-control">
-          <option *ngFor="let state of workItemStates" [value]="state">{{state}}</option>
-        </select>
-      </div>
-    </div>
-    <br /><br />
-    <div class="form-group">
-      <div class="col-sm-offset-1 col-sm-5">
-        <button
-          type="submit" class="btn" id="workItemDetail_btn_save"
-          [ngClass]="{'btn-primary': workItem.fields['system.title'],
-          'btn-primary-disabled':!workItem.fields['system.title']}"
-          (click)="save()"
-          [disabled]="workItem.fields['system.title'] ? !(workItem.fields['system.title'].trim()) : !workItem.fields['system.title']">Save</button>
-        <button type="submit" id="workItemDetail_btn_back" class="btn btn-default" (click)="goBack()">Back</button>
-      </div>
+    <h1>{{workItem.fields['system.title']}} Details</h1>
+    <br />
+    <div [hidden]="submitted">
+      <form *ngIf="active" (ngSubmit)="save()" #almWorkItemDetailForm="ngForm" id="wi-detail-form">
+        <div class="form-horizontal">
+          <div class="form-group">
+            <label class="col-sm-1 control-label">ID</label>
+            <div class="col-sm-5" id="wi-detail-id">
+              {{workItem.id}}
+            </div>
+          </div>
+          <div class="form-group">
+            <label class="col-sm-1 control-label" for="wi-detail-title">Title</label>
+            <div class="col-sm-5">
+              <input type="text" class="form-control" id="wi-detail-title"
+                [ngModel]="workItem.fields['system.title'] | almTrim "
+                (ngModelChange)="workItem.fields['system.title'] = $event"               
+                required name="title" #title="ngModel" />
```

Comment:
Nice use of the Template Reference Variable for dynamic validation. Everyone, this is how you do it. :100: